assets/js/components/ExchangeRates.js

```diff
@@ -1,106 +1,175 @@
-// ./assets/js/components/ExchangeRates.js
-
-import React, {Component} from 'react';
+import React, { useState, useEffect } from 'react';
 import axios from 'axios';
 
-class ExchangeRates extends Component {
-    constructor() {
-        super();
-        this.state = {
-            rates: {},
-            loading: true,
-            date: (new Date()).toISOString().split('T')[0]
-        };
-    }
+const ExchangeRates = () => {
+    const [rates, setRates] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [date, setDate] = useState((new Date()).toISOString().split('T')[0]);
+    const [calculateValue, setCalculateValue] = useState('');
+    const [calculateBuy, setCalculateBuy] = useState(-1);
+    const [calculateSell, setCalculateSell] = useState(-1);
+    const [calculateExchange, setCalculateExchange] = useState('');
+    const [calculateCode, setCalculateCode] = useState('');
 
-    getBaseUrl() {
-        return 'http://telemedi-zadanie.localhost';
-    }
+    const getBaseUrl = () => 'http://telemedi-zadanie.localhost';
 
-    componentDidMount() {
+    useEffect(() => {
         const urlParams = new URLSearchParams(window.location.search);
-        const date = urlParams.get('date')
-        if (date !== null) {
-            this.state.date = date
+        const dateParam = urlParams.get('date');
+        if (dateParam !== null) {
+            setDate(dateParam);
         }
+        fetchExchangeRates();
+    }, []);
 
-        this.fetchExchangeRates();
-    }
+    useEffect(() => {
+        recalculate();
+        recalculateCode();
+    }, [calculateValue, calculateBuy, calculateSell, rates]);
 
-    fetchExchangeRates() {
-        const baseUrl = this.getBaseUrl();
-        const date = this.state.date;
-        axios.get( `${baseUrl}/api/exchange-rates?date=${date}`).then(response => {
-            this.setState({
-                rates: response.data,
-                loading: false
-            })
-        }).catch(function (error) {
+    const fetchExchangeRates = () => {
+        const baseUrl = getBaseUrl();
+        axios.get(`${baseUrl}/api/exchange-rates?date=${date}`).then(response => {
```

Comment:
Myśle że dobrze by to było wstawić w nową linie, dla czytelności:
```suggestion
        axios.get(`${baseUrl}/api/exchange-rates?date=${date}`)
            .then(response => {
```